src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,271 @@
+): AsyncInterfaceFindType | undefined {
+	if (ts.isClassDeclaration(classDefinition)) {
+		const returnTypeTemplate = {
+			hasAsyncInterface: null,
+			routingAsyncFlag: null,
+			rootViewAsyncFlag: null,
+		} as AsyncInterfaceFindType;
+
+		// Checks the interfaces and manifest
+		const curClassAnalysis = classDefinition.members.reduce((acc, member) => {
+			const checkResult = doChecks(member as ts.PropertyDeclaration, manifestContent);
+			return mergeResults(acc, checkResult);
+		}, {...returnTypeTemplate});
+
+		const heritageAnalysis =
+			classDefinition?.heritageClauses?.flatMap((parentClasses: ts.HeritageClause) => {
+				return parentClasses.types.flatMap((parentClass) => {
+					const parentClassType = checker.getTypeAtLocation(parentClass);
+
+					return parentClassType.symbol?.declarations?.flatMap((declaration) => {
+						let result = {...returnTypeTemplate} as AsyncInterfaceFindType;
+						// Continue down the heritage chain to search for
+						// the async interface or manifest flags
+						if (ts.isClassDeclaration(declaration) &&
+							declaration?.name?.getText() !== "UIComponent") {
+							result = findAsyncInterface(declaration, manifestContent, checker) ?? result;
+						}
+
+						return result;
+					});
+				});
+			}) ?? [];
+
+		return [...heritageAnalysis, curClassAnalysis].reduce((acc, curAnalysis) => {
+			return mergeResults(acc ?? {...returnTypeTemplate}, curAnalysis ?? {...returnTypeTemplate});
+		});
+	}
+}
+
+function doChecks(metadata: ts.PropertyDeclaration, manifestContent: string | undefined) {
+	let classInterfaces: ts.ObjectLiteralElementLike | undefined;
+	let componentManifest: ts.ObjectLiteralElementLike | undefined;
+
+	if (metadata && ts.isPropertyDeclaration(metadata) &&
+		metadata.initializer && ts.isObjectLiteralExpression(metadata.initializer)) {
+		metadata.initializer.properties.forEach((prop) => {
+			if (prop.name?.getText() === "interfaces") {
+				classInterfaces = prop;
+			}
+			if (prop.name?.getText() === "manifest") {
+				componentManifest = prop;
+			}
+		});
+	}
+
+	let hasAsyncInterface = null;
+	if (classInterfaces && ts.isPropertyAssignment(classInterfaces) &&
+		classInterfaces.initializer && ts.isArrayLiteralExpression(classInterfaces.initializer)) {
+		hasAsyncInterface = classInterfaces.initializer
+			.elements.some((implementedInterface) =>
+				implementedInterface.getText() === "\"sap.ui.core.IAsyncContentCreation\"");
+	}
+
+	// undefined has ambiguous meaning in that context.
+	// It could mean either implicit "true" or "false".
+	// To distinguish whether it's been set from manifest's config
+	// or not set at all, we'll use null.
+	let rootViewAsyncFlag: boolean | undefined | null = null;
+	let routingAsyncFlag: boolean | undefined | null = null;
+
+	if (componentManifest && ts.isPropertyAssignment(componentManifest)) {
+		// The manifest is an external manifest.json file
+		if (componentManifest.initializer.getText() === "\"json\"") {
```

Comment:
While this is technically correct, we decided for a different handling here: `If in the Component.js no object and no "manifest: 'json'" is declared, "manifest.json" should be still analyzed`. This is because when using manifest-first, the `manifest.json` file is loaded first and is still used, even when the Component class doesn't reference it. This is obviously not correct and a reference should be maintained in the metadata. However, such cases may appear and the best practices should also be checked for them.

To improve this situation, we should add a general rule that checks for this consistency. I added it to my list.

Reply:
Thanks! Now the manifest gets always analyzed and when there's no definition for a manifest in the Component's metadata (either inline or external), there would be a warning for that